packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -659,26 +659,26 @@ abstract class RenderViewportBase<ParentDataClass extends ContainerParentDataMix
     RenderBox pivot;
     bool onlySlivers = target is RenderSliver; // ... between viewport and `target` (`target` included).
     while (child.parent != this) {
-      assert(child.parent != null, '$target must be a descendant of $this');
+      final AbstractNode parent = child.parent;
```

Comment:
Maybe do the cast to RenderObject up here instead of on line 673 (we should already know that its a RenderObject up here).

Reply:
Done